hazelcast/src/main/java/com/hazelcast/client/impl/protocol/AbstractCallableMessageTask.java

```diff
@@ -22,8 +22,8 @@
 /**
  * Base callable Message task.
  */
-public abstract class AbstractCallableMessageTask<CM extends ClientMessage>
-        extends AbstractMessageTask<CM> {
+public abstract class AbstractCallableMessageTask<P>
+        extends AbstractMessageTask<P> {
```

Comment:
Can we make Parameters extending a common interface so that we use that interface everywhere instead of generics? Because this generic is not guarding us against anything in compile time.